### ISSUE TITLE
Consider having a flag in Qualifier to control matching strictness

### ISSUE THREAD

**Issue by HansNewbie** [HubTurbo/HubTurbo#1052]
From #1050.

For milestone alias (`current`), the `Qualifier` content will be replaced with the real milestone name. However, this imposes the problem when a milestone is a substring of another since our milestone Qualifier comparison uses `contains`.

e.g. `current-10` will get `V3.1`. This will list all issues in `V3.1` and `V3.10`.

**Comment by damithc** [comment]
Yes, in some cases we need to specify an exact match. What syntax do we use? Double quotes? @dariusf

**Comment by dariusf** [comment]
Double quotes are used for qualifier contents with spaces, i.e. `milestone:"a b c"`. We could make it double up for exact matches also?

**Comment by damithc** [comment]
I'm not sure if that's a good idea. We may want to search for a string with spaces but not want an exact match. Is there a syntax we can use to specify `starts with`, `ends with`, and `exact match` in an intuitive and consistent way?

**Comment by damithc** [comment]
What if we use trailing/leading spaces?
`" aaa"` = starts with
`"aaa "` = ends with
`" aaa "` = exact match

**Comment by dariusf** [comment]
Maybe using `^` and `$` (regex anchors)? `milestone^$:"a b"` or `milestone:"^a b$"`. May be a bit cryptic though...

**Comment by damithc** [comment]
If the leading/trailing space is not workable (it is intuitive and does not look like a regex, but a bit hard to spot), then we can go for a more complex one like the one suggested.
